docs/architecture/adr-004-application-links-expiration.md

```diff
@@ -0,0 +1,249 @@
+# ADR 004: Expiration of application links
+
+## Changelog
+
+- September 20th, 2021: Initial draft;
+- September 21th, 2021: Moved from DRAFT to PROPOSED;
+- December  22th, 2021: First review;
+- January   04th, 2022: Second review.
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+Currently when a user links their centralized application with the Desmos profile, the created link contains a timestamp of when it has been created.   
+Since centralized applications' username can be switched and sold, we SHOULD implement an "expiration date" system on links. 
+This means that after a certain amount of time passes, the link will be automatically marked as invalid, and the user has to connect it again in order to keep it valid.
+
+## Context
+
+Desmos `profiles` module give the possibility to link the desmos profile with both centralized application and 
+other blockchains accounts. By doing this, a user can be verified as the owner of those accounts and prove to the system
+that they're not impersonating anyone else. This verification however remains valid only if the user
+never trade or sell his centralized-app username to someone else. If they do the link to such username MUST be invalidated. 
+Unfortunately for us, it's not possible to understand when this happens since it's off-chain. 
+To prevent this situation, an "expiration time" SHOULD be added to the centralized `ApplicationLink` object.
```

Comment:
```suggestion
Desmos `profiles` module gives users the possibility to link their Desmos profile with both centralized applications and 
other blockchains accounts. By doing this, a user can be verified as the owner of those accounts and prove to the system that they're not impersonating anyone else. This verification however remains valid only if the user
never trades or sells their centralized app username to someone else. If they do, the link to such username MUST be invalidated. 
Unfortunately for us, it's not possible to understand when this happens since it's off-chain. 
To prevent this situation, an "expiration time" SHOULD be added to the `ApplicationLink` object.
```